sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathAsyncClient.java

```diff
@@ -630,9 +637,22 @@ Mono<Response<Void>> deleteWithResponse(Boolean recursive, DataLakeRequestCondit
             .setIfModifiedSince(requestConditions.getIfModifiedSince())
             .setIfUnmodifiedSince(requestConditions.getIfUnmodifiedSince());
 
-        context = context == null ? Context.NONE : context;
-        return this.dataLakeStorage.getPaths().deleteWithResponseAsync(null, null, recursive, null, lac, mac, context)
-            .map(response -> new SimpleResponse<>(response, null));
+        // Pagination only applies to service version 2023-08-03 and later, when using OAuth.
+        Boolean paginated = (getServiceVersion().ordinal() >= DataLakeServiceVersion.V2023_08_03.ordinal()
+            && Boolean.TRUE.equals(recursive) // only applies to directories
+            && isTokenCredentialAuthenticated()) ? true : null;
+
+        Context finalContext = context == null ? Context.NONE : context;
+        return this.dataLakeStorage.getPaths()
+            .deleteWithResponseAsync(null, null, recursive, null, paginated, lac, mac, context).expand(resp -> {
```

Comment:
It might be worth adding a comment to describe what's happening here and how `expand` works (even just link to the Reactor docs).